src/app/lib/button/button.component.ts

```diff
@@ -1,16 +1,42 @@
-import { Component, EventEmitter, Output, Input } from '@angular/core';
-import { ButtonOptions } from './button-options';
+/* tslint:disable:component-selector */
+// https://github.com/mgechev/codelyzer/issues/178#issuecomment-265154480
+
+import {Component, ElementRef, Input, OnChanges, Renderer2, SimpleChanges} from '@angular/core';
+
+export function throwErrorForInvalidButtonColor() {
```

Comment:
Any reason in particular why this function exists outside of the ButtonComponent class?

Reply:
The reason for this is so that I can do unit testing to make sure that the error gets thrown appropriately